v2/tests/logger_int/main.c

```diff
@@ -0,0 +1,107 @@
+    really_big_string_fits[sizeof(really_big_string_fits) - 1] = '\0';
+
+    char really_big_string_doesnt_fit[LOG_SIZE_REGULAR]; /*if the size of this array is greater than LOG_SIZE_REGULAR than it will not get printed as all memory is statically allocated and cannot grow*/
+    (void)memset(really_big_string_doesnt_fit, 'M', sizeof(really_big_string_doesnt_fit));
+    really_big_string_doesnt_fit[sizeof(really_big_string_doesnt_fit) - 1] = '\0';
+
+    /*LogCritical*/
+    logger_log(LOG_LEVEL_CRITICAL, NULL, "LogCritical: hello world!");
+    logger_log(LOG_LEVEL_CRITICAL, NULL, "LogCritical: a more complicated hello with int=%d.", 42);
+    logger_log(LOG_LEVEL_CRITICAL, NULL, "LogCritical: the most complicated hello with int=%d and a really big string that fits =%s.", 42, really_big_string_fits);
+    logger_log(LOG_LEVEL_CRITICAL, NULL, "LogCritical: the most complicated hello with int=%d and a really big string that does not fit=%s.", 42, really_big_string_doesnt_fit);
+    logger_log(LOG_LEVEL_CRITICAL, NULL, "LogCritical: does PRI_BOOL work? %" PRI_BOOL "", MU_BOOL_VALUE(true) /*answer is yes, it works*/);
+    logger_log(LOG_LEVEL_CRITICAL, NULL, "LogCritical: does PRI_BOOL work with amazingly out of bounds value? %" PRI_BOOL "", MU_BOOL_VALUE(42));
+    logger_log(LOG_LEVEL_CRITICAL, NULL, "LogCritical: does PRI_BOOL works with false? %" PRI_BOOL "", MU_BOOL_VALUE(false));
+
+    /*LogError*/
+    logger_log(LOG_LEVEL_ERROR, NULL, "LogError: hello world!");
+    logger_log(LOG_LEVEL_ERROR, NULL, "LogError: a more complicated hello with int=%d.", 42);
+    logger_log(LOG_LEVEL_ERROR, NULL, "LogError: the most complicated hello with int=%d and a really big string that fits =%s.", 42, really_big_string_fits);
+    logger_log(LOG_LEVEL_ERROR, NULL, "LogError: the most complicated hello with int=%d and a really big string that does not fit=%s.", 42, really_big_string_doesnt_fit);
+    logger_log(LOG_LEVEL_ERROR, NULL, "LogCritical: does PRI_BOOL work? %" PRI_BOOL "", MU_BOOL_VALUE(true) /*answer is yes, it works*/);
+    logger_log(LOG_LEVEL_ERROR, NULL, "LogCritical: does PRI_BOOL work with amazingly out of bounds value? %" PRI_BOOL "", MU_BOOL_VALUE(42));
+    logger_log(LOG_LEVEL_ERROR, NULL, "LogCritical: does PRI_BOOL works with false? %" PRI_BOOL "", MU_BOOL_VALUE(false));
+
+//#ifdef _MSC_VER /*linux doesn't have LogLastError*/
+//    /*LogLastError*/
+//    LogLastError("LogLastError: hello world!");
+//    LogLastError("LogLastError: a more complicated hello with int=%d.", 42);
+//    LogLastError("LogLastError: the most complicated hello with int=%d and a really big string that fits =%s.", 42, really_big_string_fits);
+//    LogLastError("LogLastError: the most complicated hello with int=%d and a really big string that does not fit=%s.", 42, really_big_string_doesnt_fit);
+//#endif
+    /*LogWarning*/
+    logger_log(LOG_LEVEL_WARNING, NULL, "LogWarning: hello world!");
```

Comment:
>LogWarning [](http://example.com/codeflow?start=41&length=10)

I think all these LogWarning should dissapear from here, as the level is given as argument to the function,

Reply:
Removed indeed from here and the old int tests